R/tar-stars.R

```diff
@@ -204,63 +204,49 @@ tar_stars_raw <- function(name,
 
     driver <- rlang::arg_match0(driver, drv$name)
 
-    .read_stars <- eval(substitute(
-        function(path) {
-
-        ## TODO: it appears envvar custom resources do not work in read function?
-        READ_FUN <- stars::read_stars
-        #  mdim <- as.logical(Sys.getenv("GEOTARGETS_GDAL_RASTER_MDIM", unset = FALSE))
-        if (mdim) {
-            READ_FUN <- stars::read_mdim
-        }
-
-        # ncdf <- as.logical(Sys.getenv("GEOTARGETS_USE_NCMETA", unset = FALSE))
-        if (ncdf && requireNamespace("ncmeta")) {
-            READ_FUN <- stars::read_ncdf
-        }
-
-        # proxy <- as.logical(Sys.getenv("GEOTARGETS_PROXY", unset = FALSE))
-        READ_FUN(path, proxy = proxy)
-
-    }, list(ncdf = ncdf, mdim = mdim, proxy = proxy)))
-
-    # TODO: should multidimensional array use the same `options` as 2D?
-    .write_stars <- function(object, path) {
-
-        WRITE_FUN <- stars::write_stars
-
-        mdim <- as.logical(Sys.getenv("GEOTARGETS_GDAL_RASTER_MDIM",
-                                      unset = FALSE))
-        if (mdim) {
-            WRITE_FUN <- stars::write_mdim
-        }
-
-        dr <- Sys.getenv("GEOTARGETS_GDAL_RASTER_DRIVER")
-
-        # stars requires character(0), not "", for no options set
-        co <- Sys.getenv("GEOTARGETS_GDAL_RASTER_CREATION_OPTIONS")
-        co2 <- strsplit(co, ";")[[1]]
-
-        WRITE_FUN(
-            object,
-            path,
-            overwrite = TRUE,
-            driver = dr,
-            options = co
-        )
-    }
-
     targets::tar_target_raw(
         name = name,
         command = command,
         pattern = pattern,
         packages = packages,
         library = library,
         format = targets::tar_format(
-            read = .read_stars,
-            write = .write_stars,
-            marshal = function(object) object, # Not currently used
-            unmarshal = function(object) object
+            read = function(path) {
+                if (ncdf && requireNamespace("ncmeta")) {
```

Comment:
I did this, but unfortunately because it's inside of `tar_format()` I have to use `geotargets:::check_pkg_installed()` which gives a 'Note' on check.  I think that's fine though and we can explain it to CRAN